### PR TITLE
Update eslint and plugins

### PR DESCRIPTION
**Pull request by mroderick**
This PR does three things:

* upgrade `eslint` to latest
* use `eslint-config-sinon` instead of in-repository config of rules
* add `eslint-plugin-ie11` to detect incompatibility with IE11

### CODE REVIEW

**Comment by mroderick** [comment]
> Could/should we somehow bring back a compatibility table in normal Sinon?

I am not sure I understand what you're suggesting. Could you rephrase to help me understand?

**Comment by fatso83** [comment]
@mroderick I totally understand you don't understand. It was a non-sensical question: we already have this at the top of the README. I just looked at the homepage.
